dsls/sql/src/test/java/org/apache/beam/dsls/sql/BeamSqlDslBase.java

```diff
@@ -62,11 +69,37 @@ public static void prepareClass() throws ParseException {
 
     recordsInTableA = prepareInputRecordsInTableA();
 
-    inputA1 = PBegin.in(pipeline).apply("inputA1", Create.of(recordsInTableA)
-        .withCoder(new BeamSqlRowCoder(recordTypeInTableA)));
+    boundedInput1 = PBegin.in(pipeline).apply("boundedInput1",
+        Create.of(recordsInTableA).withCoder(new BeamSqlRowCoder(recordTypeInTableA)));
+
+    boundedInput2 = PBegin.in(pipeline).apply("boundedInput2",
+        Create.of(recordsInTableA.get(0)).withCoder(new BeamSqlRowCoder(recordTypeInTableA)));
+
+    unboundedInput1 = prepareUnBoundedPCollection1();
+    unboundedInput2 = prepareUnBoundedPCollection2();
+  }
+
+  private static PCollection<BeamSqlRow> prepareUnBoundedPCollection1() {
+    TestStream.Builder<BeamSqlRow> values = TestStream
+        .create(new BeamSqlRowCoder(recordTypeInTableA));
+
+    for (BeamSqlRow row : recordsInTableA) {
+      values = values.advanceWatermarkTo(new Instant(row.getDate(7)));
```

Comment:
Can you add getTYPE(String fieldName) overloads to BeamSqlRow similar to getFieldValue(), and change this to getDate("f_timestamp")? The integer indices are quite opaque.